plugins/nomad-backend/src/service/router.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { errorHandler, requestLoggingHandler } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import { InputError } from '@backstage/errors';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+
+/** @public */
+export interface RouterOptions {
+  logger: Logger;
+  config: Config;
+}
+
+/** @public */
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { config, logger } = options;
+
+  // Get Nomad addr and token from config
+  const addr = config.getString('nomad.addr');
+  const token = config.getOptionalString('nomad.token');
+
+  const router = Router();
+  router.use(express.json());
+
+  router.get('/health', (_, resp) => {
+    resp.json({ status: 'ok' });
+  });
+
+  router.get('/v1/allocations', async (req, res) => {
+    // Check namespace argument
+    const namespace = (req.query.namespace as string) ?? '';
+    if (!namespace || namespace === '') {
+      throw new InputError(`Missing "namespace" query parameter`);
+    }
+
+    // Check filter argument
+    const filter = (req.query.filter as string) ?? '';
+    logger.debug(`request headers: namespace=${namespace} filter=${filter}`);
+
+    // Issue the request
+    const endpoint = `${addr}/v1/allocations?namespace=${encodeURIComponent(
+      namespace,
+    )}&filter=${encodeURIComponent(filter)}`;
+    const allocationsResp = await fetch(endpoint, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'X-Nomad-Token': token || '',
+      },
+    });
+
+    // Check response
+    if (allocationsResp.status !== 200) {
+      const body = await allocationsResp.text();
+      logger.error(`failed to call /v1/allocations endpoint: ${body}`);
+      res.status(allocationsResp.status).send(body);
```

Comment:
Make sure that all `.send()` are replaced with res.json instead. See https://github.com/backstage/backstage/blob/master/SECURITY.md#express-responses for more info

Reply:
Sounds good to me, sorry I missed that!